service/controller/v24/adapter/cpf.go

```diff
@@ -1,25 +1,13 @@
 package adapter
 
-import (
-	"context"
-
-	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
-	"github.com/giantswarm/microerror"
-
-	"github.com/giantswarm/aws-operator/service/routetable"
-)
-
 // CPFConfig represents the config for the adapter collection for the Control
 // Plane Finalizer management.
 type CPFConfig struct {
-	RouteTable *routetable.RouteTable
-
-	AvailabilityZones          []v1alpha1.AWSConfigStatusAWSAvailabilityZone
 	BaseDomain                 string
 	ClusterID                  string
-	EncrypterBackend           string
 	GuestHostedZoneNameServers string
-	NetworkCIDR                string
+	PrivateRoutes              []CPFRouteTablesRoute
+	PublicRoutes               []CPFRouteTablesRoute
 	Route53Enabled             bool
 }
```

Comment:
Do we even need to use `*Config`? Can't we just name it `CPF` and use it directly as a pure data struct?

Reply:
I am planning to do this. I also want to move the data structure to the templates. Will do with the next PRs.